src/main/java/com/newestuser/imen/bgdate/BulgarianDate.java

```diff
@@ -147,7 +147,7 @@ private String findYearType(int bulgarianDateYear) {
      * @return true if the passed year is leap.
      */
     private static boolean findIsLeapYear(int currentYear) {
-        return currentYear % 4 == 0 && currentYear % 100 != 0 || currentYear % 400 == 0;
```

Comment:
Actually some correction needs to be done.

The year 1800 is not a leap year. You can read more on that [here](https://docs.microsoft.com/en-us/office/troubleshoot/excel/determine-a-leap-year).

Write a test regarding the year 1800 and the year 2020. 1800 should not be calculated as a leap year and the year 2020 should be calculated as a leap year.